#56 Canvas helper to add hw from iiif image info

rlskoeser opened the pull request:
implements `set_hwd_from_iiif` helper as described in #56 

I created a placeholder `set_hwd` for testing purposes; maybe this should not be merged until the other helper is implemented

- any other cases to handle with the image url passed in?
- wasn't sure what was desired for error handling; should we just document possible exceptions?
- I had to add `requests` to tox dependencies to get tests to work locally, shouldn't it pull from setup.py?

NOTE: this work depends on #70 . Once that is merged, this can be updated and the placeholder `set_hwd` method removed.

digitaldogsbody commented:
> I created a placeholder `set_hwd` for testing purposes; maybe this should not be merged until the other helper is implemented

Agree, since the dummy implementation will need to be removed and it would be nicer to wait and remove that before merge.

>     * any other cases to handle with the image url passed in?
Not that I can immediately think of

>     * wasn't sure what was desired for error handling; should we just document possible exceptions?
I think that raising the errors to the main library is fine - we can document that the helper may raise and leave handling that up to the end user. I think this is better than silently failing and essentially requiring them to check for success after calling the function.

>     * I had to add `requests` to tox dependencies to get tests to work locally, shouldn't it pull from setup.py?
Afraid I'm not sure about this. @helrond, @p-galligan - any ideas?

rlskoeser commented:
I'll set this to draft until the other helper has been merged

giacomomarchioro commented on iiif_prezi3/helpers/canvas_helpers.py:
```diff
@@ -3,21 +3,21 @@
+                an Annotation and ResourceItem attached.
+        """
         body = ResourceItem(id=image_url, type='Image')
         annotation = Annotation(id=anno_id, body=body, target=image_url, motivation='painting', type='Annotation')
```
Just a comment, the type of the main IIIF object should be set by default from the JSON schema, so theoretically there should be no need for specifying it for Annotation and AnnotationPage, Canavses etc. etc.

giacomomarchioro reviewed (approved):
Impressive work, I think we should also think if it is the case of allowing to add directly a service using the `add_image` method.

rlskoeser commented:
Earlier today the PR diff was showing a lot of changes I didn't make — I thought I had updated my fork and rebased my branch properly, not sure why it was flagging so many other changes. I just updated again and now the diff is accurate. Sorry about that.

digitaldogsbody commented:
> Earlier today the PR diff was showing a lot of changes I didn't make — I thought I had updated my fork and rebased my branch properly, not sure why it was flagging so many other changes. I just updated again and now the diff is accurate. Sorry about that.

I think it merged rather than rebasing on top of the changes, because the commit list was showing a lot of commits from main branch.

Thanks for the great work - I'll merge based on the two approvals and follow the action instructions for getting the GH Pages setup working :)
